docs/develop/porting.rst

```diff
@@ -0,0 +1,308 @@
+.. _porting_to_a_board:
+
+Porting MicroPython to a New Board
```

Comment:
Technically this is porting to a new "architecture" or "microcontroller family"

Conceptually, ports are like "STM32" or "nRF" or "ESP32" ... etc  (confusingly STM32 and nRF are both the same architecture, and STM32 contains multiple "families").

However, "board" is already a well defined concept -- a port contains multiple boards which are like instances of the port for a specific piece of hardware that you can actually buy. And although defining a new board configuration is a common thing to do, that's not what this section is describing.

So perhaps suggest rename this section to just "Porting MicroPython" and maybe a small re-word of these first two paragraphs:

```
The MicroPython project contains several ports to different microcontroller families and architectures. The project repository has a `ports <https://github.com/micropython/micropython/tree/master/ports>`_
directory containing a subdirectory for each supported port.

A port will typically contain definitions for multiple "boards", each of which is a specific piece of hardware that that port can run on (i.e. a development kit or device).

The
`minimal port <https://github.com/micropython/micropython/tree/master/ports/minimal>`_ is available as simplified reference implementation of a MicroPython port. In general, starting a port requires

- Setting up the toolchain (configuring Makefiles, etc)
- Implementing boot configuration and CPU initialisation
- Initialising basic drivers required for development and debugging (e.g. GPIO, UART).
- Performing the board-specific configurations.
- Implementing the port-specific modules.
```